### PR TITLE
Issue #199927: Fix terminal tab list drag/drop selection

### PR DESCRIPTION
**Pull request by MNouel**
Issue #199927: Dragging inactive terminal task tab does not work properly

Description: Forcibly sets the dragged tab to be selected if the target instance (the tab it is getting dragged onto) was previously selected before the drag. This fixes the issue where the background color of the previously selected tab remained as the selected background color while the actually active tab was another.

Testing: Open two or more tabs, select one of them, and then drag the inactive tab over to the active tab. The tag you dragged will now become active/selected with appropriate UI.

Video: https://github.com/microsoft/vscode/assets/143637929/7e0afcff-eaba-440b-8170-692b4cd0307d

### CODE REVIEW

**Comment by MNouel** [comment]
@microsoft-github-policy-service agree

File: src/vs/workbench/contrib/terminal/browser/terminalTabsList.ts [hunk 1]
```diff
@@ -707,10 +708,27 @@ class TerminalTabsDragAndDrop extends Disposable implements IListDragAndDrop<ITe
 		}
 
 		let focused = false;
+		// Retrieve the currently active instance asynchronously
+		let activeInstancePromise = this._terminalService.activeInstance;
+		let setActiveInstance = async (instance: ITerminalInstance) => {
+			const activeInstance = await activeInstancePromise; // Ensure the promise is resolved before comparison.
```
Comment (Tyriar) [inline src/vs/workbench/contrib/terminal/browser/terminalTabsList.ts:714]: `this._terminalService.activeInstance` is syunc, so this await doesn't seem to do anything.

File: src/vs/workbench/contrib/terminal/browser/terminalTabsList.ts [hunk 2]
```diff
@@ -707,10 +708,27 @@ class TerminalTabsDragAndDrop extends Disposable implements IListDragAndDrop<ITe
...
 			if (!focused) {
-				this._terminalService.setActiveInstance(instance);
+				// Await the setActiveInstance call to keep the loop synchronous
+				await setActiveInstance(instance);
+				if (targetWasFocus) {
+					if (targetIndex)
+						this._listService.lastFocusedList?.setSelection([targetIndex]);
+					else
+						this._listService.lastFocusedList?.setSelection([0]);
+				}
```
Comment (Tyriar) [inline src/vs/workbench/contrib/terminal/browser/terminalTabsList.ts:731]: This fixes the problem when dragging the second, inactive tab to the top. But not when dragging the first, inactive tab to the bottom.

File: src/vs/workbench/contrib/terminal/browser/terminalTabsList.ts [hunk 3]
```diff
@@ -707,10 +708,27 @@ class TerminalTabsDragAndDrop extends Disposable implements IListDragAndDrop<ITe
 		}
 
 		let focused = false;
+		// Retrieve the currently active instance asynchronously
+		let activeInstancePromise = this._terminalService.activeInstance;
```
Comment (Tyriar) [inline src/vs/workbench/contrib/terminal/browser/terminalTabsList.ts:712]: There's a bunch of lint warnings about using const instead of let

**Review by Tyriar — CHANGES_REQUESTED** [review]

**Pull request closed without merging**